scripts/fabtests/ccm_runfabtests.sh

```diff
@@ -0,0 +1,196 @@
+      p) provider="$OPTARG";;
+      t) test_suite="-t $OPTARG";;
+      u) use_parameters=1;;
+      v) verbose="${verbose} -v";;
+      *) usage 1;;
+    esac
+  done
+fi
+
+#
+# Check for ccmlogin
+#
+ccmlogin=`command -v ccmlogin`
+if [ $? != 0 ]; then
+    ${load_ccm_module}
+    if [ $? != 0 ]; then
+        echo "Error: CCM Module not loaded.  Can Not execute the FAB tests."
+        exit 1
+    fi
+
+    ccmlogin=`command -v ccmlogin`
+    if [ $? != 0 ]; then
+        echo "Error: CCM Login command not found.  Can Not execute the FAB tests."
+        exit 1
+    fi
+fi
+
+rm -f ${ccm_login_script}
+cat > ${ccm_login_script} <<"END_OF_CCMLOGIN_SCRIPT_1"
+#!/bin/bash
+
+END_OF_CCMLOGIN_SCRIPT_1
+
+echo "debug=${debug}" >> ${ccm_login_script}
+echo "fabtests_directory='${fabtests_directory}'" >> ${ccm_login_script}
+echo "provider='${provider}'" >> ${ccm_login_script}
+echo "test_suite='${test_suite}'" >> ${ccm_login_script}
+echo "use_parameters='${use_parameters}'" >> ${ccm_login_script}
+echo "verbose='${verbose}'" >> ${ccm_login_script}
+
+cat >> ${ccm_login_script} <<"END_OF_CCMLOGIN_SCRIPT_2"
+node_1=""
+node_2=""
+
+if [[ "${provider}" == "gni" ]]
+then
+    ccm_login_node_dir="${HOME}/.crayccm/"
+    ccm_nodelist_file="${ccm_login_node_dir}/`ls ${ccm_login_node_dir}`"
+    node_index=0
+
+    if [[ ${debug} -ne 0 ]]
+    then
+        echo "*** pid: $$, ccm_nodelist_file: '${ccm_nodelist_file}'"
+    fi
+
+    while read -r node_entry
+    do
+        if [[ ${debug} -ne 0 ]]
+        then
+            echo "*** pid: $$, next node_index: '${node_index}'";
+            echo "*** pid: $$, next node_entry: '${node_entry}'";
+        fi
+        if [[ ${node_index} -eq 0 ]]
+        then
+            if [[ ${use_parameters} -eq 0 ]]
+            then
+                node_1="${node_entry}"
+            else
+                node_1="-s ${node_entry}"
+            fi
+            node_index=1
+            if [[ ${debug} -ne 0 ]]
+            then
+                echo "*** pid: $$, set node_1: '${node_1}'";
+            fi
+        else
+            if [[ "${node_2}" == "" ]]
+            then
+                if [[ "${node_1}" != "${node_entry}" ]]
+                then
+                    if [[ ${use_parameters} -eq 0 ]]
+                    then
+                        node_2="${node_entry}"
+                    else
+                        node_2="-c ${node_entry}"
+                    fi
+                    if [[ ${debug} -ne 0 ]]
+                    then
+                        echo "*** pid: $$, set node_2: '${node_2}'";
+                    fi
+                    break
+                fi
+            fi
+        fi
+    done <${ccm_nodelist_file}
+
+    if [[ ${debug} -ne 0 ]]
+    then
+        echo "*** pid: $$, final node_1: '${node_1}'"
+        echo "*** pid: $$, final node_2: '${node_2}'"
+    fi
+fi
+
+echo "*** cmd: 'cd ${fabtests_directory}'"
+cd ${fabtests_directory}
+
+if [[ ${use_parameters} -eq 0 ]]
```

Comment:
I'm glad you got both versions working.  Let's leave it in here for now in case things change with runfabtests.sh, but eventually, let's just use the `use_parameters` path.